spec/lib/extensions/ar_to_model_hash_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "spec_helper"
+
+describe ToModelHash do
+  class TestVm < ActiveRecord::Base
+    has_one        :test_hardware, :dependent => :destroy
+    virtual_column :bitness,   :type => :integer, :uses => :test_hardware
+    virtual_column :num_disks, :type => :integer, :uses => {:test_hardware => :test_disks}
+  end
+
+  class TestHardware < ActiveRecord::Base
+    has_many   :test_disks
+    belongs_to :test_vm
+  end
+
+  class TestDisk < ActiveRecord::Base
+    belongs_to :test_hardware
+  end
+
+  let(:test_to_model_hash_options) do
+    {
+      "cols"    => ["name"],
+      "include" =>
+        {"test_hardware" =>
+          {
+            "columns" => ["bitness"],
+            "include" => {"test_disk" => {"columns" => ["num_disks"]}}
```

Comment:
I don't understand how this could pass...the model defines it plural, but here and on line 50 it's singular?